### ISSUE TITLE
Important label blocked by unimportant icon (for example amenity=university vs private parking icon)

### ISSUE THREAD

**Issue by matkoniecz** [gravitystorm/openstreetmap-carto#964]
from #287 

http://www.openstreetmap.org/#map=19/41.86037/12.49663
http://www.openstreetmap.org/relation/1517075

![przechwytywanie](https://cloud.githubusercontent.com/assets/899988/4345376/32d3d574-40ca-11e4-8149-9d0e285e3cca.PNG)

http://www.openstreetmap.org/way/257147754

http://www.openstreetmap.org/?mlat=50.81602&mlon=12.93069#map=19/50.81602/12.93069
http://www.openstreetmap.org/way/24518628

http://www.openstreetmap.org/?mlat=50.81602&mlon=12.93069#map=18/41.86018/12.49709

**Comment by mkoniecz** [comment]
I propose to move parking icon to .amenity-low-priority layer - it should fix this problem.

**Comment by math1985** [comment]
This is an issue that is being reported repeatedly, but it's not clear what the best solution is. Does anybody have a suggestion?

**Comment by matkoniecz** [comment]
Move more icons/labels to .amenity-low-priority layer? Or create .amenity-high-priority layer for some labels (university, school)?

Or sort objects by area - I think that it is already done for some features.

**Comment by math1985** [comment]
Moving icons to low-priority might improve the situation (for parking and playground for example), but some problems will remain. A small shop in a hospital might still block the hospital label for example.

> Or sort objects by area

Objects are sorted by area, but that's not sufficient as icons always come before labels.

**Comment by HolgerJeromin** [comment]
and some things are nodes with no area.

**Comment by matkoniecz** [comment]
@HolgerJeromin 
> some things are nodes with no area

What indicates low importance. And in areas where mapping is just starting and universities are mapped as nodes it is unlikely that there is big problem with high feature density.

**Comment by kocio-pl** [comment]
I think this is more general problem: on high zoom levels I'm expecting to see all the small features, while when zooming out I expect to loose details and see only bigger objects.

**Comment by jeisenbe** [comment]
I believe this problem still exists because all icons from `.points` are rendered before `.text` labels. This means that if there is any feature with an icon rendered, such as a parking lot or a drinking fountain, it will block the text label for a large feature, such as a University.

This also happens with features that are rendered with both an icon and a label. In that case, and important text label may be blocked by another minor feature located just south of the important feature. 

This might be resolved by creating a new layer for important text labels which would render before all icons, or by merging the amenity point and text layers and using a priority ranking list to determine rendering order for icons and text together.

(BTW, does anyone know of an issue that describes this problem more generally? I think we've discussed in comments on other issues as well. Related issues include #1026 - where it is mentioned that some text labels disappear or reappear when zooming in)
